contracts/6/forwarder/ERC20ForwardRequestCompatible.sol

```diff
@@ -4,15 +4,19 @@ pragma experimental ABIEncoderV2;
 contract ERC20ForwardRequestTypes{
 
     struct ERC20ForwardRequest {
-        address from; //index
-        address to; //index
-        address token; //index
+        address from; 
+        address to; 
+        address token; 
         uint256 txGas;
         uint256 tokenGasPrice;
-        uint256 batchId; //maybe
-        uint256 batchNonce; //maybe
-        uint256 deadline;
+        uint256 batchId; 
+        uint256 batchNonce; 
+        uint256 deadline; 
         bytes data;
     }
     
-}
\ No newline at end of file
+}
+
+// token gas price is : ( gas price * 10^ token decimals ) / (token prince in USD / ETH price in USD)
```

Comment:
Either add this comment at top in contract level comments.. or someplace else... this is unorganised

Reply:
Agreed. Done